### ISSUE TITLE
Azure Function 3608 error

### ISSUE THREAD

**Issue by bszewczyk** [Azure/Azure-DataFactory#234]
Azure Functions block failes after 4 minutes of working despite azure function returns 200 code.

Error:
Error code
3608
Troubleshooting guide
Failure type
User configuration issue
Details
Call to provided Azure function 'DecomposeData_AzureFunction_DecomposeDataWithPreds' failed with status-'BadGateway' and message - '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"> <html xmlns="http://www.w3.org/1999/xhtml"> <head> <meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1"/> <title>502 - Web server received an invalid response while acting as a gateway or proxy server.</title> <style type="text/css">  </style> </head> <body> <div id="header"><h1>Server Error</h1></div> <div id="content"> <div class="content-container"><fieldset> <h2>502 - Web server received an invalid response while acting as a gateway or proxy server.</h2> <h3>There is a problem with the page you are looking for, and it cannot be displayed. When the Web server (while acting as a gateway or proxy) contacted the upstream content server, it received an invalid response from the content server.</h3> </fieldset></div> </div> </body> </html> '.

**Comment by CharlesAtDCube** [comment]
Same here

**Comment by bszewczyk** [comment]
Yes, I thought that is some problem with time that my functions last. I thought that maybe it can be fixed by Microsoft

**Comment by bszewczyk** [comment]
Why issue is closed? That means it is fixed or won't be fixed?

**Comment by fhljys** [comment]
@bszewczyk , I thought Charles provided a solution though it is not a good one. But I will reopen it and notify related people. Thanks.

**Comment by fhljys** [comment]
Also find this: https://docs.microsoft.com/en-us/azure/data-factory/data-factory-troubleshoot-guide#error-code-3608

**Comment by bszewczyk** [comment]
Hi,

like it is written here https://docs.microsoft.com/en-us/azure/data-factory/control-flow-azure-function-activity#timeout-and-long-running-functions

Azure Fucntion block can last maximum 230 seconds even though funcion can last longer. My question is if it will be improved in azure data factory.

**Comment by fhljys** [comment]
That document actually links to here: https://docs.microsoft.com/en-us/azure/azure-functions/functions-versions#timeout
![image](https://user-images.githubusercontent.com/18674958/111300580-348aff00-868c-11eb-93b4-7ff1f51ffd5e.png)
So the limitation is actually from Azure Function instead of Azure Data Factory, right?

**Comment by CharlesAtDCube** [comment]
Hello,

Since the 230s limit is related to the fact that we call Azure Functions via HTTP, with RESTful verbs : the http calls should be stateless. As it is supposed to be stateless, it's legitimate that azure function blocks should not last for a long time.
So the good practice would be to have a stateless function that returns "202 Accepted" or "201 Created" (job) for the time of the job run and after the job is done, the answer becomes "200 {"status": "succeded", "data" : [...]}.

[cid:ab5215ad-6eaa-4a27-8eaa-74867dabe9d9]<https://dcube.fr/>
-
Charles GOSSELIN
***@***.******@***.***>
-
170, rue Raymond Losserand
75014 Paris
________________________________
De : Beata Szewczyk ***@***.***>
Envoyé : mardi 16 mars 2021 12:11
À : Azure/Azure-DataFactory ***@***.***>
Cc : Charles GOSSELIN ***@***.***>; Comment ***@***.***>
Objet : Re: [Azure/Azure-DataFactory] Azure Function 3608 error (#234)


Hi,

like it is written here https://docs.microsoft.com/en-us/azure/data-factory/control-flow-azure-function-activity#timeout-and-long-running-functions

Azure Fucntion block can last maximum 230 seconds even though funcion can last longer. My question is if it will be improved in azure data factory.
